oshmem/mca/sshmem/ucx/configure.m4

```diff
@@ -29,33 +29,19 @@ AC_DEFUN([MCA_oshmem_sshmem_ucx_CONFIG],[
     save_CPPFLAGS="$CPPFLAGS"
 
     alloc_dm_LDFLAGS=" -L$ompi_check_ucx_libdir/ucx"
-    alloc_dm_LIBS=" -luct_ib"
     CPPFLAGS+=" $sshmem_ucx_CPPFLAGS"
     LDFLAGS+=" $sshmem_ucx_LDFLAGS $alloc_dm_LDFLAGS"
-    LIBS+=" $sshmem_ucx_LIBS $alloc_dm_LIBS"
-
-    AC_LANG_PUSH([C])
-    AC_LINK_IFELSE([AC_LANG_PROGRAM(
-          [[
-            #include <ucp/core/ucp_resource.h>
-            #include <uct/ib/base/ib_alloc.h>
-          ]],
-          [[
-            uct_md_h md = ucp_context_find_tl_md((ucp_context_h)NULL, "");
-            (void)uct_ib_md_alloc_device_mem(md, NULL, NULL, 0, "", NULL);
-            uct_ib_md_release_device_mem(NULL);
-          ]])],
-          [
-           AC_MSG_NOTICE([UCX device memory allocation is supported])
-           AC_DEFINE([HAVE_UCX_DEVICE_MEM], [1], [Support for device memory allocation])
-           sshmem_ucx_LIBS+=" $alloc_dm_LIBS"
-           sshmem_ucx_LDFLAGS+=" $alloc_dm_LDFLAGS"
-          ],
-          [
-           AC_MSG_NOTICE([UCX device memory allocation is not supported])
-           AC_DEFINE([HAVE_UCX_DEVICE_MEM], [0], [Support for device memory allocation])
-          ])
-    AC_LANG_POP([C])
+    LIBS+=" $sshmem_ucx_LIBS"
+
+    AC_CHECK_DECL([UCS_MEMORY_TYPE_RDMA], 
+    [
+        AC_MSG_NOTICE([UCX device memory allocation is supported])
+        AC_DEFINE([HAVE_UCX_DEVICE_MEM], [1], [Support for device memory allocation])
```

Comment:
not needed. AC_CHECK_DECLS (notice the S) already defines macros, check online docs..